### PR TITLE
Opensearch+Metadata DB

### PR DESCRIPTION
**Pull request by ianmcorvidae**
This includes all the OpenSearch-related changes for the time being, including moving tag indexing to this service instead of it happening in cyverse-de/monkey, which can now be removed.

This service still only does periodic indexing and does not handle CyVerse metadata, so there remains work to be done, but this is the base thing that's needed.

### CODE REVIEW

File: dedb.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+// BeginTx starts an DEDBTx for the given DEDBConnection
+func (d *DEDBConnection) BeginTx(ctx context.Context, opts *sql.TxOptions) (*DEDBTx, error) {
+	tx, err := d.db.BeginTx(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	stats := d.db.Stats()
+	log.Infof("DEDB stats: %d/%d open; %d/%d used/idle", stats.OpenConnections, stats.MaxOpenConnections, stats.InUse, stats.Idle)
```
Comment (slr71) [inline dedb.go:60]: 👍

File: main.go [hunk 2]
```diff
@@ -180,7 +195,7 @@ func publishPrefixMessages(context context.Context, prefixes []string, client *m
 		if err != nil {
 			rejectErr := del.Reject(!del.Redelivered)
 			if rejectErr != nil {
-				log.Error(rejectErr)
+				log.Error(errors.Wrap(rejectErr, "Failed rejecting the index message after failing to publish prefix messages"))
```
Comment (slr71) [inline main.go:198]: 👍

**Review by slr71 — APPROVED** [review]
I didn't catch any problems. 👍

**Review by psarando — APPROVED** [review]
I didn't catch any issues 👍

**Pull request merged**